src/services/api/index.tsx

```diff
@@ -29,61 +31,87 @@ const MAX_SUGGESTIONS = 50;
 
 /**
  * Sends PATCH request to api.
- * @param {string} endpoint - URL endpoint.
+ * @param endpoint - URL endpoint.
  * @param {Object} payload - PATCH payload.
  */
-const patch = (endpoint, payload, callOptions) => {
+const patch = (endpoint: string, payload: Record<string, unknown>): Promise<GeneralRecordType> => {
   // strip out the display name to have it force-regenerate
   const { displayName, ...changes } = payload;
   const init = {
     method: 'PATCH',
     body: jc.stringify(changes),
   };
-  return request(endpoint, init, callOptions);
+  return request(endpoint, init);
 };
 
 /**
  * Sends GET request to api.
  * @param {string} endpoint - URL endpoint.
  */
-const get = (endpoint, callOptions) => {
+const get = (endpoint: string, callOptions?: RequestCallOptions) => {
   const init = {
     method: 'GET',
   };
   return request(endpoint, init, callOptions);
 };
 
 /**
- * Sends POST request to api.
+ * Sends POST request to api to create a new record.
  * @param {string} endpoint - URL endpoint.
  * @param {Object} payload - POST payload.
  */
-const post = (endpoint, payload, callOptions) => {
+function post<Resp = unknown>(endpoint: string, payload?: Record<string, unknown>): Promise<Resp> {
```

Comment:
You should be able to use the general record type here i think as well

Reply:
the `'/license/sign'` endpoint is called with this function. does that return a general record?

Reply:
yes it should return the updated user record

Reply:
[63dd5ad](https://github.com/bcgsc/pori_graphkb_client/pull/52/commits/63dd5ad5b12fbda6593e780f761b742b6c007dc4)